### PR TITLE
Json depth limits

### PR DESCRIPTION
**Pull request by BrainHorse**
Read and wrote arbitrarily deep JSON. Reasons for a change:

1. Running out of stack is not a failure a caller can handle. The codec recurses once per level of nesting, so a deep enough document exhausts the stack.
2. The depth that breaks it was a property of the thread, not the document. The frame budget is whatever stack the calling thread was given. The same document read on two threads succeeded on one and overflowed on the other. A document that passes in one service could fail in the service that reads it. A fixed limit makes the outcome depend on the input alone.
3. Writing deeper than play-json reads produced documents nobody could consume. At depth 1001 the
codec wrote clean bytes that its own reader accepted and `Json.parse` refused.

Note: play-json's own writer has no depth limit either, so `Json.toBytes` produces documents `Json.parse`
refuses - its own output, unchanged. This codec is now stricter than play-json there: it refuses to
write past the depth play-json can read. The only documents that refusal costs are ones no play-json
reader could have consumed, and the error arrives at the writer instead of in whatever service
parses it later. Callers who genuinely need deeper output can build a codec with a higher bound.

Performance cost is negligible. The recorded `post 1.4` benchmark figures still stand ± a few %.

## Summary by CodeRabbit

* **New Features**
  * Added configurable nesting-depth limits for JSON value encoding and decoding.
  * Introduced a default maximum depth of 1,000, with supported limits up to 10,000.
  * JSON processing now reports clear failures when configured nesting limits are exceeded.

* **Bug Fixes**
  * Improved handling of deeply nested arrays and objects to prevent stack-depth failures.

* **Documentation**
  * Clarified that serialization may fail when nesting exceeds supported limits and that partial output may already have been written.

* **Tests**
  * Added coverage for depth limits, boundary values, custom configurations, and invalid settings.

### CODE REVIEW

File: play-json-jsoniter/shared/src/main/scala/play/api/libs/json/JsonValueCodecJsValue.scala [hunk 1]
```diff
@@ -35,9 +40,29 @@ object JsonValueCodecJsValue {
...
       bigDecimalSerializerSettings: BigDecimalSerializerConfig
+  ): JsonValueCodec[JsValue] =
+    unsafe(bigDecimalParseSettings, bigDecimalSerializerSettings, DefaultMaxNestingDepth)
+
+  def of(
+      bigDecimalParseSettings: BigDecimalParseConfig,
+      bigDecimalSerializerSettings: BigDecimalSerializerConfig,
+      maxNestingDepth: Int
+  ): Either[String, JsonValueCodec[JsValue]] =
+    if (maxNestingDepth > 0) {
```
Comment (stasimus) [inline play-json-jsoniter/shared/src/main/scala/play/api/libs/json/JsonValueCodecJsValue.scala:51]: Only the lower end is validated. of(..., Int.MaxValue) is accepted and brings back the stack overflow this PR removes. Cap it, or say so in the scaladoc.

File: build.sbt [hunk 2]
```diff
@@ -126,6 +126,7 @@ lazy val benchmark = project
   .enablePlugins(JmhPlugin)
   .settings(
     commonSettings,
+    allowUnsafeScalaLibUpgrade := true,
```
Comment (stasimus) [inline build.sbt:129]: Unrelated to depth limits. Is it needed? benchmark is publish/skip with Mima disabled.

**Review by stasimus — APPROVED** [review]

File: build.sbt (hunk 2 above)
Comment (BrainHorse) [reply build.sbt:129]: I was unable to run benchmarks without this.
```shell
[error] (benchmark / scalaInstance) Expected `benchmark / scalaVersion` to be 2.13.18 or later, but found 2.13.16.
```
This is indeed unrelated to the main change, but I decided not to create a separate PR for this single line change.

File: play-json-jsoniter/shared/src/main/scala/play/api/libs/json/JsonValueCodecJsValue.scala (hunk 1 above)
Comment (BrainHorse) [reply play-json-jsoniter/shared/src/main/scala/play/api/libs/json/JsonValueCodecJsValue.scala]: Done, with an explanation of the numbers in doc
